src/storage/cloud-extensions.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Bucket,
+  BucketOptions,
+  File,
+  FileOptions,
+  Storage as StorageClient,
+} from "@google-cloud/storage";
+interface FirebaseMetadata {
+  name: string;
+  bucket: string;
+  generation: string;
+  metageneration: string;
+  contentType: string;
+  timeCreated: string;
+  updated: string;
+  storageClass: string;
+  size: string;
+  md5Hash: string;
+  contentEncoding: string;
+  contentDisposition: string;
+  crc32c: string;
+  etag: string;
+  downloadTokens: string;
+}
+export class FirebaseStorageFile extends File {
+  // Gets metadata from firebase backend instead of GCS
+  getFirebaseMetadata(): Promise<FirebaseMetadata> {
+    // We need to talk to the firebase storage endpoints instead of the google cloud bucket endpoint
+    const endpoint = "https://firebasestorage.googleapis.com/v0";
```

Comment:
How can configure this for testing against our internal testing endpoints or emulator?

Reply:
The expectation is that you call `getStorage(app, 'myendpoint')` or `getStorage(undefined, 'myEndpoint')`